server/src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -309,227 +287,6 @@ public static int getId(Class<? extends ElasticsearchException> exception) {
-
-        for (Map.Entry<String, List<String>> entry : metadata.entrySet()) {
-            headerToXContent(builder, entry.getKey().substring("es.".length()), entry.getValue());
-        }
-
-        if (throwable instanceof ElasticsearchException) {
-            ElasticsearchException exception = (ElasticsearchException) throwable;
-            exception.metadataToXContent(builder, params);
-        }
-
-        if (params.paramAsBoolean(REST_EXCEPTION_SKIP_CAUSE, REST_EXCEPTION_SKIP_CAUSE_DEFAULT) == false) {
-            if (cause != null) {
-                builder.field(CAUSED_BY);
-                builder.startObject();
-                generateThrowableXContent(builder, params, cause);
-                builder.endObject();
-            }
-        }
-
-        if (headers.isEmpty() == false) {
-            builder.startObject(HEADER);
-            for (Map.Entry<String, List<String>> entry : headers.entrySet()) {
-                headerToXContent(builder, entry.getKey(), entry.getValue());
-            }
-            builder.endObject();
-        }
-
-        if (params.paramAsBoolean(REST_EXCEPTION_SKIP_STACK_TRACE, REST_EXCEPTION_SKIP_STACK_TRACE_DEFAULT) == false) {
-            builder.field(STACK_TRACE, Exceptions.stackTrace(throwable));
-        }
-
-        Throwable[] allSuppressed = throwable.getSuppressed();
-        if (allSuppressed.length > 0) {
-            builder.startArray(SUPPRESSED.getPreferredName());
-            for (Throwable suppressed : allSuppressed) {
-                builder.startObject();
-                generateThrowableXContent(builder, params, suppressed);
-                builder.endObject();
-            }
-            builder.endArray();
-        }
-    }
-
-    private static void headerToXContent(XContentBuilder builder, String key, List<String> values) throws IOException {
-        if (values != null && values.isEmpty() == false) {
-            if (values.size() == 1) {
-                builder.field(key, values.get(0));
-            } else {
-                builder.startArray(key);
-                for (String value : values) {
-                    builder.value(value);
-                }
-                builder.endArray();
-            }
-        }
-    }
-
-    /**
-     * Renders additional per exception information into the XContent
-     */
-    protected void metadataToXContent(XContentBuilder builder, Params params) throws IOException {
-    }
-
-    /**
-     * Generate a {@link ElasticsearchException} from a {@link XContentParser}. This does not
-     * return the original exception type (ie NodeClosedException for example) but just wraps
-     * the type, the reason and the cause of the exception. It also recursively parses the
-     * tree structure of the cause, returning it as a tree structure of {@link ElasticsearchException}
-     * instances.
-     */
-    public static ElasticsearchException fromXContent(XContentParser parser) throws IOException {
```

Comment:
is this safe to remove? (bwc consideration).

Reply:
Yes, it was only used by itself (recursively) and by `DefaultShardOperationFailedException.fromXContent` - which itself is also never used.

This was even unrelated to the storage changes and could've been removed before.